dashboard/lib/client_state.rb

```diff
@@ -62,42 +62,36 @@ def level_progress_is_empty_for_test
   def add_script(script_id)
     s = scripts
     s << script_id.to_i unless s.include?(script_id)
-    session[:scripts] = s
+    cookies[:scripts] = JSON.generate(s)
   end
 
   # Returns an array of ids of the scripts completed in the current session.
   # Callers should not mutate the array.
   # @return [Array<Integer>]
   def scripts
-    session[:scripts] || []
-  end
-
-  # Returns a read-only set of the videos seen in the current user session,
-  # for tests only.
-  # @return Set<String>
-  def videos_seen_for_test
-    session[:videos_seen] || Set.new
+    parse_json(cookies[:scripts])
   end
 
   # Adds video_key to the set of videos seen in the current user session.
   # @param [String] video_key
   def add_video_seen(video_key)
-    (session[:videos_seen] ||= Set.new).add(video_key)
+    v = videos_seen
+    v << video_key unless v.include?(video_key)
+    cookies[:videos_seen] = JSON.generate(v)
   end
 
-  # Adds video_key to the set of videos seen in the current user session.
+  # Returns true if the given video has been seen.
   # @param [String] video_key
   # @return Boolean
   def video_seen?(video_key)
-    s = session[:videos_seen]
-    s && s.include?(video_key)
+    videos_seen.include?(video_key)
   end
 
   # Returns if at least one video has been seen in the current user session.
   # For testing only
   # @return Boolean
   def videos_seen_for_test?
```

Comment:
Not exactly.

We could keep the naming convention, make the method private and then use send to invoke it in tests, which would be a bad code smell in non-test code.